crates/biome_js_analyze/src/lint/nursery/no_bitwise.rs

```diff
@@ -0,0 +1,191 @@
+use serde::{Deserialize, Serialize};
+
+declare_lint_rule! {
+    /// Disallow bitwise operators
+    ///
+    /// The use of bitwise operators in JavaScript is very rare and often & or | is simply a mistyped && or ||,
+    /// which will lead to unexpected behavior.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// let x = y | z;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// x |= y;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// let x = y || z;
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// The rule provides the option described below.
+    ///
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "allow": ["&", "|", "^", "~", "<<", ">>", ">>>"],
+    ///         "int32Hint": false,
+    ///     }
+    /// }
+    /// ```
+    /// ### allow
+    ///
+    /// Allows a list of bitwise operators to be used as exceptions.
+    ///
+    /// Examples of valid code for this rule with the `{ "allow": ["~"] }` option:
+    ///
+    /// ```js
+    /// ~[1,2,3].indexOf(1) === -1;
+    /// ```
+    /// ### int32Hint
+    ///
+    /// Allows the use of bitwise OR in |0 pattern for type casting.
+    ///
+    /// Default: `false`
+    ///
+    /// Examples of valid code for this rule with the `{ "int32Hint": true }` option:
+    ///
+    /// ```js
+    /// const b = a|0;
+    /// ```
+    ///
+    pub NoBitwise {
+        version: "next",
+        name: "noBitwise",
+        language: "js",
+        sources: &[
+            RuleSource::Eslint("no-bitwise"),
+        ],
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub AnyExpressionWithBitwise = JsBinaryExpression | JsUnaryExpression | JsAssignmentExpression
+}
+
+impl AnyExpressionWithBitwise {
+    fn operator_token(&self) -> Option<JsSyntaxToken> {
+        match self {
+            AnyExpressionWithBitwise::JsBinaryExpression(node) => node.operator_token().ok(),
+            AnyExpressionWithBitwise::JsUnaryExpression(node) => node.operator_token().ok(),
+            AnyExpressionWithBitwise::JsAssignmentExpression(node) => node.operator_token().ok(),
+        }
+    }
+    fn exist_bitwise_op(&self) -> bool {
+        match self {
+            AnyExpressionWithBitwise::JsBinaryExpression(_) => {
+                self.operator_token().is_some_and(|op| {
+                    matches!(op.kind(), T![&] | T![|] | T![^] | T![<<] | T![>>] | T![>>>])
+                })
+            }
+            AnyExpressionWithBitwise::JsUnaryExpression(_) => self
+                .operator_token()
+                .is_some_and(|op| matches!(op.kind(), T![~])),
+            AnyExpressionWithBitwise::JsAssignmentExpression(_) => {
+                self.operator_token().is_some_and(|op| {
+                    matches!(
+                        op.kind(),
+                        T![&=] | T![|=] | T![^=] | T![<<=] | T![>>=] | T![>>>=]
```

Comment:
I believe, in our `biome_js_syntax` we have enums that match these tokens. If possible, we should use those enums:

https://github.com/biomejs/biome/blob/21dd358be5dade01d672916a0ab76bc825ef85cb/crates/biome_js_syntax/src/expr_ext.rs#L177

The file contains more utilities that might help here